File: Practica_4_java/src/Logica/FiguraGeometrica.java

package Logica;



public class FiguraGeometrica {

    
    public double calcularArea(double radio) {
        return Math.PI * radio * radio;
    }

    
    public double calcularArea(double base, double altura) {
        return base * altura;
    }

    
    public double calcularArea(double base, double altura, boolean esTriangulo) {
        if (esTriangulo) {
            return (base * altura) / 2;
        }
        return 0;
    }

    
    public double calcularArea(double baseMayor, double baseMenor, double altura) {
        return ((baseMayor + baseMenor) * altura) / 2;
    }

    
    public double calcularArea(double lado, double apotema, String penta) {
        if (penta=="si") {
           return (5 * lado * apotema) / 2;
        }
           return 0;
       
    }

    public static void main(String[] args) {
        FiguraGeometrica fg = new FiguraGeometrica();

        
        double areaCirculo = fg.calcularArea(5);
        System.out.println("Área del círculo: " + areaCirculo);

        
        double areaRectangulo = fg.calcularArea(4, 6);
        System.out.println("Área del rectángulo: " + areaRectangulo);

        
        double areaTriangulo = fg.calcularArea(4, 6, true);
        System.out.println("Área del triángulo rectángulo: " + areaTriangulo);

        
        double areaTrapecio = fg.calcularArea(8, 4, 5);
        System.out.println("Área del trapecio: " + areaTrapecio);

        
        double areaPentagono = fg.calcularArea(6, 4,"si");
        System.out.println("Área del pentágono: " + areaPentagono);
    }
}

File: Practica_4_python/FiguraGeometrica.py
import math


class Geometria:

    
    def calcular_area(self, *args):
        if len(args) == 1:  
            radio = args[0]
            return math.pi * radio * radio  
        elif len(args) == 2:  
            base, altura = args
            return base * altura  

        elif len(args) == 3:  
            base, altura, divisor = args
            return (base * altura) / divisor  



        elif len(args) == 4:  
            base_mayor, base_menor, altura, divisor = args
            return ((
                                base_mayor + base_menor) * altura) / 2  
        elif len(args) == 5:  
            apotema, perimetro, esPoligono, divisor, noPoligono = args
            return (5 * perimetro * apotema) / divisor 
        else:
            raise ValueError("Número de parámetros incorrecto para calcular el área")



geometria = Geometria()


print("Área del círculo: ", geometria.calcular_area(5))


print("Área del rectángulo: ", geometria.calcular_area(4, 6))


print("Área del triángulo rectángulo: ", geometria.calcular_area(4, 6, 2))


print("Área del trapecio: ", geometria.calcular_area(8, 4, 5, 2))


print("Área del pentágono: ", geometria.calcular_area(6, 4, "si", 2, "falso"))
